pontoon/base/middleware.py

```diff
@@ -83,3 +79,55 @@ def __call__(self, request):
 
         request.session["next_path"] = request.get_full_path()
         return redirect(email_consent_url)
+
+
+class ThrottleIpMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.max_count = settings.THROTTLE_MAX_COUNT
+        self.observation_period = settings.THROTTLE_OBSERVATION_PERIOD
+        self.block_duration = settings.THROTTLE_BLOCK_DURATION
+
+        # Set to block_duration if longer, otherwise requests will be blocked until
+        # the observation_period expires rather than when the block_duration expires
+        if self.observation_period > self.block_duration:
+            self.observation_period = self.block_duration
+
+    def __call__(self, request):
+        if settings.THROTTLE_ENABLED is False:
+            return self.get_response(request)
+
+        ip = get_ip(request)
+
+        # Generate cache keys
+        throttle_key = f"throttle_ip_{ip}"
```

Comment:
nitpick: I would go with `throttled_key` for consistency on these (throttled/blocked).